Build failure: kdeltachat

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *kdeltachat*

### Build log

```
error: builder for '/nix/store/pblmnw96v9ab73yl500d2dg8wjrks8pz-kdeltachat-unstable-2023-01-31.drv' failed with exit code 2;
       last 10 log lines:
       >  2956 | dc_accounts_t* dc_accounts_new                  (const char* dir, int writable);
       >       |                                                                   ~~~~^~~~~~~~
       > [ 69%] Building CXX object CMakeFiles/kdeltachat.dir/qml_HtmlViewSheet_qml.cpp.o
       > [ 71%] Building CXX object CMakeFiles/kdeltachat.dir/qml_ConfigurePage_qml.cpp.o
       > [ 73%] Building CXX object CMakeFiles/kdeltachat.dir/qml_main_qml.cpp.o
       > [ 75%] Building CXX object CMakeFiles/kdeltachat.dir/qmlcache_loader.cpp.o
       > make[2]: *** [CMakeFiles/kdeltachat.dir/build.make:208: CMakeFiles/kdeltachat.dir/accounts.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:101: CMakeFiles/kdeltachat.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/pblmnw96v9ab73yl500d2dg8wjrks8pz-kdeltachat-unstable-2023-01-31.drv'.
```

### Additional context

nixpkgs [rev](https://github.com/NixOS/nixpkgs/commit/dd5621df6dcb90122b50da5ec31c411a0de3e538) (recent unstable). fyi i hadn't tried this package before, so not sure if the issue is recent.

### Notify maintainers

@dotlambda

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0-zen2, NixOS, 24.05 (Uakari), 24.05.20240114.dd5621d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(kiara): `"home-manager"`
 - channels(root): `""`
 - nixpkgs: `/nix/store/dllvbhgw5f584ficlqvqkpb0mjnf4v0s-source`
```

--- a/pkgs/applications/networking/instant-messengers/kdeltachat/default.nix
+++ b/pkgs/applications/networking/instant-messengers/kdeltachat/default.nix
@@ -14,13 +14,13 @@
 
 mkDerivation rec {
   pname = "kdeltachat";
-  version = "unstable-2023-01-31";
+  version = "unstable-2024-01-14";
 
   src = fetchFromSourcehut {
     owner = "~link2xt";
     repo = "kdeltachat";
-    rev = "0c9370cfe41ae7f99b4fceced896f66fb4e9195c";
-    hash = "sha256-6KSzsPs8tSzVOxGUWj/AvSJihrSwamZgUNGvjnmNnag=";
+    rev = "d61a01c2d6d5bdcc9ca500b466ed42689b2bd5c6";
+    hash = "sha256-KmL3ODXPi1c8C5z2ySHg0vA5Vg/dZumDZTbpxkzf7A4=";
   };
 
   nativeBuildInputs = [

